internal/core/swrenderer/draw_functions.rs

```diff
@@ -418,6 +418,8 @@ impl TargetPixel for Rgb565Pixel {
         // gggggggg_000rrrrr_rrr000bbb_bbbbb00
         let c =
             ((color.red as u32) << 13) | ((color.green as u32) << 24) | ((color.blue as u32) << 2);
+        // gggggg00_000rrrrr_00000bbb_bb00000
+        let c = c & 0b11111100_00011111_00000111_11000000;
```

Comment:
I don't completely get the math behind this and I haven't really tried what it does visually, but I came across this while trying to understand this to implement some Gbr555.

The comments at line 418 and 421 only have 7 digit for the last byte, and I was wondering if, consequently, the blue bits mask shouldn't be shifted right by one bit so that there are 5x unset bits on the right (for self.blue) and 6x unset bits on the left (for the blended green bits).

Else my understanding is that the `res >> 5` at the end would leave the blue LSB always unset and the blue MSB would overlap with the green LSB.

Reply:
Looks like the comments are indeed wrong.
I think I computed the thing so that the most significant bit gets to the right place. I hope I didn't loose too much precision on the LSB.

Reply:
Nice catch! there was indeed a one bit off in there.
Fixed in commit b620986